creusot/tests/should_succeed/swap_borrows.mlcfg

```diff
@@ -68,20 +68,20 @@ module SwapBorrows_Swap
     goto BB1
   }
   BB1 {
-    assume { Resolve0.resolve _3 };
     _3 <- (let (_, a) = x_1 in a);
     x_1 <- (let (a, b) = x_1 in (a, any t));
-    assume { Resolve0.resolve _4 };
     _4 <- (let (a, _) = x_1 in a);
     x_1 <- (let (a, b) = x_1 in (any t, b));
-    assume { Resolve1.resolve x_1 };
+    assume { Resolve0.resolve x_1 };
     _0 <- (_3, _4);
     goto BB2
   }
   BB2 {
+    assume { Resolve1.resolve _4 };
```

Comment:
As far as I understand `_4` and `_3` should be unitialized, so why do we resolve them here?

Reply:
After the assignment, `_3` and `_4` are uninitialized and dead but then we drop them which makes them become initialized again.
<details><summary>MIR</summary>

```rust
bb1
  _0 = (move _3, move _4) [_1] -> [_1]
    live=[_3, _4] -> [_0] init=[_1, _3, _4] -> [_0, _1] uninit=[_0, _2] -> [_2, _3, _4]
  drop(_4) -> [return: bb2, unwind: bb5] [_1] -> [_1, _4]
    live=[_0] -> [_0] init=[_0, _1] -> [_0, _1, _4] uninit=[_2, _3, _4] -> [_2, _3]
bb2
  StorageDead(_4) [_1, _4] -> [_1]
    live=[_0] -> [_0] init=[_0, _1, _4] -> [_0, _1] uninit=[_2, _3] -> [_2, _3, _4]
  drop(_3) -> [return: bb3, unwind: bb6] [_1] -> [_1, _3]
    live=[_0] -> [_0] init=[_0, _1] -> [_0, _1, _3] uninit=[_2, _3, _4] -> [_2, _4]
bb3
  StorageDead(_3) [_1, _3] -> [_1]
    live=[_0] -> [_0] init=[_0, _1, _3] -> [_0, _1] uninit=[_2, _4] -> [_2, _3, _4]
```

</details>

Reply:
Wait... On the contrary, dropping a variable makes it UNinitialized, right? We consume its content!

Reply:
The operand of drop counts as mutating use (similar to the `&mut self` argument to the drop function).

Reply:
Who does this choice? To me, the operand is passed by-move, so the variable should be considered as consumed after the drop.

The drop method of the Drop trait indeed takes a mutable borrow, but this is different from the drop function, which takes value by full ownership.

Reply:
Do you mean it makes `_3` and `_4` *live*? Because that's true, but it should definitely de initialize them

Reply:
Something is fishy here, according to the output you posted earlier in the thread, it shouldn't be the case we resolve `_3` and `_4`. they are in both maybe initialized and maybe uninitalized so they should be ruled out from consideration. since `init = maybe init - maybe uninit`.

Reply:
This mir code makes me so confused, why are we even calling `drop` on `_4`? is this something related to drop elaboration?